src/core/cgroup.c

```diff
@@ -740,19 +746,27 @@ static void cgroup_context_apply(Unit *u, CGroupMask mask, ManagerState state) {
         if ((mask & CGROUP_MASK_MEMORY) && !is_root) {
                 if (cg_unified() > 0) {
                         uint64_t max = c->memory_max;
+                        uint64_t swap_max = c->memory_swap_max;
 
-                        if (cgroup_context_has_unified_memory_config(c))
+                        if (cgroup_context_has_unified_memory_config(c)) {
                                 max = c->memory_max;
-                        else {
+                                swap_max = c->memory_swap_max;
+                        } else {
                                 max = c->memory_limit;
 
                                 if (max != CGROUP_LIMIT_MAX)
                                         log_cgroup_compat(u, "Applying MemoryLimit %" PRIu64 " as MemoryMax", max);
+
+                                swap_max = c->memory_swap_limit;
+
```

Comment:
systemd has already two memory control options.
`MemoryMax=bytes` is for `"memory.max"` unified hierarchy control group attribute. `MemoryLimit=bytes` is for `"memory.limit_in_bytes"` legacy control group attribute.

Now, I'd like to add two memory options, `MemorySwapMax=` and `MemorySwapLimit=`. Do you want to add only `MemorySwapMax=` option?
Many of systems are still using legacy cgroup. So, to support legacy, I think both options are needed. Further, there are difference in memory operations between legacy and unified. At legacy, `memory.memsw.limit_in_bytes` is accounting and limiting memory+swap. But, at unified, `memory.swap.max` handles only swap.

So, I think, systemd needs both options to make a pair of corresponding memory control option.